### PR TITLE
Fail if more scenarios fail

### PR DESCRIPTION
**Pull request by jhowlett-scottlogic**
PR for issue: https://github.com/ScottLogic/openapi-forge/issues/87

- Save test results from workflow as a .json file.
- Add step that retrieves test results .json file from most recent push to master. 
- Add step which runs a .js script that for each language compares the two files.
- Fails if languages exist in both json files and:
    - more tests in new run: newFailedTotal - oldFailedTotal > numberOfAddedTests
    - less tests in new run:  newFailedTotal - oldFailedTotal > 0
    - This allows for tests to be added and not cause the testing to fail.
- Fix bug that would result in a non-zero exit code when tests pass

Please excuse the obscene number of commits and degeneration of commit messages. To develop I had to create a commit to trigger the workflow to run.

This will set up the generator state webpage nicely https://github.com/ScottLogic/openapi-forge/issues/69 as all that will be needed is a new workflow that updates the webpage .md file with the new tests results from the json.

### CODE REVIEW

**Review by ColinEberhardt — CHANGES_REQUESTED** [review]
a few minor comments, but otherwise looks really good.

**Comment by ColinEberhardt** [comment]
> Please excuse the obscene number of commits and degeneration of commit messages. To develop I had to create a commit to trigger the workflow to run.

No worries - a squash will fix that 👍

File: src/index.js [hunk 1]
```diff
@@ -60,6 +60,10 @@ program
     "Sets the logging level, options are: quiet ('quiet', 'q' or '0'), standard (default) ('standard', 's' or '1'), verbose ('verbose', 'v' or '2')",
     "1"
   )
+  .option(
+    "-r, --reportResults",
```
[the remark this answers is not in the document]
Comment (jhowlett-scottlogic) [reply src/index.js:64]: Ahh yes that is in the forge command too

File: src/index.js [hunk 2]
```diff
@@ -60,6 +60,10 @@ program
     "Sets the logging level, options are: quiet ('quiet', 'q' or '0'), standard (default) ('standard', 's' or '1'), verbose ('verbose', 'v' or '2')",
     "1"
   )
+  .option(
```
Comment (ColinEberhardt) [inline src/index.js:63]: Could this be changed to ...

~~~
.option(
    "-o, --outputFile <file>",
    "Writes the testing results to a JSON file",
    "test-results.json"
  )
~~~

Comment (jhowlett-scottlogic) [reply src/index.js:63]: I have made a slight tweak to the implementation of this requested change. The reason is that this proposed solution would result in a file always being generated. With my version only when the argument `-o` is given does it write an output. If no path is given it uses the default (test-results.json) and if given it uses the given path.

**Review by ColinEberhardt — APPROVED** [review]

**Pull request merged**